lib/comparable/diff.ex

```diff
@@ -106,34 +106,20 @@ defmodule Timex.Comparable.Diff do
       do_diff_months(start_date, end_date) * -1
     end
   end
-  defp do_diff_months({y,m,_}, {y,m,_}), do: 0
+  defp do_diff_months({y, m, _}, {y, m, _}), do: 0
   defp do_diff_months({y1, m1, d1}, {y2, m2, d2}) when y1 <= y2 and m1 < m2 do
     year_diff = y2 - y1
-    month_diff = if d2 >= d1, do: m2 - m1, else: (m2-1)-m1
-    (year_diff*12)+month_diff
+    month_diff = if d2 >= d1, do: m2 - m1, else: (m2 - 1) - m1
+    (year_diff * 12) + month_diff
```

Comment:
I did some style adjustments also, hope you don't mind...

Reply:
@kelvinst 

I like the following logic:

```
(year(d1) * 12 + month(d1)) - (year(d2) * 12 + month(d2))
```

It does not use if logic, which I think is a positive thing.

Reply:
Well, I do prefer it too, but this way we would not be considering the day of the month on the diff, and I particularly **need** to consider the day on my use case.

I'll try to think how we can still consider the day and get a better code on this. Thanks for the suggestion.